src/main/java/org/generationcp/middleware/dao/NameDAO.java

```diff
@@ -493,6 +475,35 @@ public boolean checkIfMatches(final String name) {
 		return map;
 	}
 
+	public List<String> getNamesByGidsAndPrefixes(final List<Integer> gids, final List<String> prefixes) {
+		try {
+			final StringBuilder sql = new StringBuilder();
+			sql.append("SELECT nval FROM names WHERE gid IN (:gids) ")
+				.append(" AND ( ");
+
+			final int prefixesSize = prefixes.size();
+			for (int i=0; i<prefixesSize; i++) {
```

Comment:
We can replace this whole for loop by:
`final List<String> formattedPrefixes = prefixes.stream().map(prefix -> " nval LIKE '" + prefix + "'").collect(Collectors.toList());
		sql.append(StringUtils.join(formattedPrefixes, " OR "));`